mlfinlab/feature_importance/importance.py

```diff
@@ -0,0 +1,144 @@
+        for j in X.columns:
+            X1_ = X.iloc[test, :].copy(deep=True)
+            np.random.shuffle(X1_[j].values)  # Permutation of a single column
+            if scoring == 'neg_log_loss':
+                prob = fit.predict_proba(X1_)
+                features_metrics_values.loc[i, j] = -scoring_func(y.iloc[test], prob, sample_weight=sample_weight[test],
+                                                              labels=clf.classes_)
+            else:
+                pred = fit.predict(X1_)
+                features_metrics_values.loc[i, j] = scoring_func(y.iloc[test], pred,
+                                                                   sample_weight=sample_weight[test])
+
+    imp = (-features_metrics_values).add(fold_metrics_values, axis=0)
+    if scoring == 'neg_log_loss':
+        imp = imp / -features_metrics_values
+    else:
+        imp = imp / (1. - features_metrics_values)
+    imp = pd.concat({'mean': imp.mean(), 'std': imp.std() * imp.shape[0] ** -.5}, axis=1)
+    return imp
+
+
+def feature_importance_sfi(clf, X, y, cv_gen, sample_weight=None, scoring='neg_log_loss'):
+    """
+    Snippet 8.4, page 118. Implementation of SFI
+
+    This function generates Single Feature Importance based on OOS score (using cross-validation object)
+    :param clf: (sklearn.ClassifierMixin): any sklearn classifier
+    :param X: (pd.DataFrame): train set features
+    :param y: (pd.DataFrame, np.array): train set labels
+    :param cv_gen: (PurgedKFold): cross-validation object
+    :param sample_weight: (np.array): sample weights, if None equal to ones
+    :param scoring: (str): scoring function used to determine importance
+    :return: (pd.DataFrame): mean and std feature importance
+    """
+    feature_names = X.columns
+    if sample_weight is None:
+        sample_weight = np.ones((X.shape[0],))
+
+    imp = pd.DataFrame(columns=['mean', 'std'])
+    for feat in feature_names:
+        feat_cross_val_scores = ml_cross_val_score(clf, X=X[[feat]], y=y, sample_weight=sample_weight,
+                                                   scoring=scoring, cv_gen=cv_gen)
+        imp.loc[feat, 'mean'] = feat_cross_val_scores.mean()
+        imp.loc[feat, 'std'] = feat_cross_val_scores.std() * feat_cross_val_scores.shape[0] ** -.5
+    return imp
+
+
+def plot_feature_importance(imp, oob_score, oos_score, savefig=False, output_path=None):
+    """
+    Snippet 8.10, page 124. Feature importance plotting function
+
+    Plot feature importance function
+    :param imp: (pd.DataFrame): mean and std feature importance
+    :param oob_score: (float): out-of-bag score
+    :param oos_score: (float): out-of-sample (or cross-validation) score
+    :param savefig: (bool): boolean flag to save figure to a file
+    :param output_path: (str): if savefig is True, path where figure should be saved
+    :return: None
+    """
+    # Plot mean imp bars with std
+    plt.figure(figsize=(20, 10))
```

Comment:
Was there a reason for not using de Prado's adaptive figure sizing? Alternatively, you could make figsize an argument, just so the option is available.

Reply:
Very good point! Changed that